client/src/main/java/io/confluent/kafka/schemaregistry/client/CachedSchemaRegistryClient.java

```diff
@@ -38,9 +38,9 @@
 
   private final RestService restService;
   private final int identityMapCapacity;
-  private final Map<String, Map<Schema, Integer>> schemaCache;
+  private final Map<String, IdentityLRUCache<Schema, Integer>> schemaCache;
   private final Map<String, Map<Integer, Schema>> idCache;
```

Comment:
is there a reason we don't need to expire idCache entries?

Reply:
IDs are supposed to be fixed, modulo deleting the entire subject. I only replaced the cases where we previously had capped-sized `IdentityHashMap`s. For the `idCache` there wasn't any limit and no `IdentityHashMap`s.

We probably should constrain the total size here with a cache (especially given the naming of the field...), but here I'm just trying to impose proper caching behavior on the existing fixed-sized maps that seem to cause problems. The `idCache` sub-maps seem to be unconstrained at the moment, and without problems that we've seen in practice.

But I agree, all the fields in this class that are acting as caches should end up labeled as such. Even better if we can abstract out a generic `Cache` interface instead of `Map`, and label these fields generically and then allocate a good concrete type for them. But atm, I figured getting correct behavior trumps all the right internal types.

Reply:
idCache used to be size-constrained in some cases as a byproduct of our existing limit. This may not be the case anymore and might cause problems (see my comment below)